https://github.com/tlvu/jupyterhub
git diff 5bcbc8b328d24eb18deb4ca9b82b13d633522d76 11a85d1dc545a5e6295a607d5d09c4b026b3c3a9 share/jupyterhub/templates/login.html
PR https://github.com/jupyterhub/jupyterhub/pull/3264

diff --git a/share/jupyterhub/templates/login.html b/share/jupyterhub/templates/login.html
index 9359d9ec..6a946d66 100644
--- a/share/jupyterhub/templates/login.html
+++ b/share/jupyterhub/templates/login.html
@@ -68,6 +68,18 @@
         <i class="fa fa-spinner"></i>
       </div>
     </div>
+
+    {% block login_terms %}
+    {% if login_term_url %}
+    <div id="login_terms" class="login_terms">
+    <input type="checkbox" id="login_terms_checkbox" name="login_terms_checkbox"/>
+    {% block login_terms_text %} {# allow overriding the text #}
+    By logging into the platform you accept the <a href="{{ login_term_url }}">terms and conditions</a>.
+    {% endblock login_terms_text %}
+    </div>
+    {% endif %}
+    {% endblock login_terms %}
+
   </div>
 </form>
 {% endif %}
@@ -92,4 +104,24 @@ $('form').submit((e) => {
   form.find('.feedback-widget>*').toggleClass('fa-pulse');
 });
 </script>
+
+{% block login_terms_script %} {# allow overriding the javascript #}
+{% if login_term_url %}
+<script>
+// By default disable the login submit button, unless user check the terms and
+// conditions checkbox.
+var loginBtn = document.getElementById('login_submit');
+loginBtn.disabled = "disabled";
+
+document.getElementById('login_terms_checkbox').onclick = function() {
+    if ( this.checked ) {
+        loginBtn.disabled = "";
+    } else {
+        loginBtn.disabled = "disabled";
+    }
+};
+</script>
+{% endif %}
+{% endblock login_terms_script %}
+
 {% endblock %}
